README.md

```diff
@@ -53,30 +53,51 @@ Will validate TypeScript code in the project. This requires a `tsconfig.json` fi
 
-1. `master` – ongoing development
+1. `trunk` – ongoing development
 1. `alpha` – release candidate
+1. `hotfix/*` - for testing urgent bugfixes
+1. `epic/*` - for testing large-scale features
 1. `release` – the production-ready, released code
 
 The following assumes that CI will run:
 
-1. `npm run release` for `release`, `alpha`, and `hotfix/*` branches
+1. `npm run release` for `release`, `alpha`, `hotfix/*`, and `epic/*` branches
 1. `post-release.sh` script on `release` branch, after the above command completes
 
 ### Regular release flow
 
-1. Commit ongoing changes to `master` branch, using [structured commit messages](https://www.conventionalcommits.org/en/v1.0.0/)
-1. Merge `master` into `alpha` to create a release candidate (e.g. `1.2.0-alpha.1`)
-1. Merge `alpha` into `release` to create a release (e.g. `1.2.0`)
-1. `alpha` branch will be reset on top of `release`
-1. `master` branch will be updated with the changes from the `release` branch
+1. Create a new `feat/*` or `fix/*` (or `chore/*` or `style/*`) branch off the `trunk` branch.
+1. Commit changes to your branch using [structured commit messages](https://www.conventionalcommits.org/en/v1.0.0/).
+1. Open a pull request for review based on `trunk`. Changes must be tested and approved before merging.
+1. Merge approved changes to the `trunk` branch. When merging into `trunk`, SQUASH the merge.
+1. Merge `trunk` into `alpha` to create a release candidate (e.g. `1.2.0-alpha.1`). When merging `trunk` into `alpha`, DO NOT SQUASH the merge.
+1. Merge `alpha` into `release` to create a production release (e.g. `1.2.0`). When merging `alpha` into `release`, DO NOT SQUASH the merge.
+1. `alpha` branch will be reset on top of `release`.
+1. `trunk` branch will be updated with the changes from the `release` branch.
+
+### Epic feature release flow
+
+For large-scale features that require more than one interdependent branch throughout development.
+
+1. Create a new `epic/*` branch off the `trunk` branch. Push the branch to GitHub so all engineers can work off it simultaneously. **Use `git pull origin trunk` to keep this branch up-to-date with `trunk`, to minimize the risk of merge conflicts.**
+1. For new features or fixes, create new `feat/*` or `fix/*` (or `chore/*` or `style/*`) branches off the epic branch. **Use `git pull origin epic/<branch-name>` to keep your feature and fix branches up-to-date with the `epic/*` branch, to minimize the risk of merge conflicts.**
+1. Commit changes to your feature or fix branches using [structured commit messages](https://www.conventionalcommits.org/en/v1.0.0/).
+1. Open pull requests for review based on the `epic/*` branch. Changes must be tested and approved before merging.
+1. Merge approved changes to the `epic/*` branch. When merging into `epic/*`, DO NOT SQUASH the merge.
```

Comment:
Why is that? We avoid squashing when merging `trunk -> alpha` or `alpha -> release` to retain commit history because of the changelog generation, but this does not apply here.

Reply:
We ran into issues with squashing commits into the epic when we were working on the Campaigns Rearchitecture project. Several of us were all working on sub-branches from a single epic branch. We'd open PRs against this epic branch, and then squash the commits when merging to it as usual. But after that, when everyone else pulled down those merged changes to their local copies of the epic and tried to merge those updates into their own in-progress sub-branches, Git lacked the full commit history and wouldn't know which changes it should keep, so it ask us to resolve a huge number of merge conflicts, even when it was clear to us which changes should be kept.

We found that by merging without squashing into the epic, the epic branch retains a full commit history, which allows everyone to `git pull` to the epic and keep their own sub-branches up-to-date with the shared history without creating merge conflicts. Instead, it creates a clean auto-merge with a merge commit and generally no conflicts. When we finally merge the epic to `trunk`, we do squash, so all the additional merge commits generated by this workflow won't make it back to the main branch.

Reply:
FWIW, we took this non-squashing approach for the `epic/ras-acc` branches and it's worked quite well. It allowed Laurel and Miguel to work simultaneously on pretty huge changesets that touched many of the same files without worrying which one would be merged first. Then, when Miguel's got merged first, it was easy to pull those changes into Laurel's branch without generating a lot of conflicts.

Reply:
Thank you for the detailed reply, this makes sense!